nautobot_secrets_providers/providers/delinea.py

```diff
@@ -197,41 +195,41 @@ def query_thycotic_secret_server(  # pylint: disable=too-many-boolean-expression
                 os.environ["REQUESTS_CA_BUNDLE"] = original_env
 
 
-class ThycoticSecretServerSecretsProviderId(ThycoticSecretServerSecretsProviderBase):
-    """A secrets provider for Thycotic Secret Server."""
+class DelineaSecretServerSecretsProviderId(DelineaSecretServerSecretsProviderBase):
+    """A secrets provider for Delinea Secret Server."""
 
-    slug = "thycotic-tss-id"  # type: ignore
-    name = "Thycotic Secret Server by ID"  # type: ignore
+    slug = "delinea-tss-id"  # type: ignore
+    name = "Delinea Secret Server by ID"  # type: ignore
 
     # TBD: Remove after pylint-nautobot bump
     # pylint: disable-next=nb-incorrect-base-class
     class ParametersForm(BootstrapMixin, forms.Form):
-        """Required parameters for Thycotic Secret Server."""
+        """Required parameters for Delinea Secret Server."""
 
         secret_id = forms.IntegerField(
             label="Secret ID",
             required=True,
             min_value=1,
-            help_text="The secret-id used to select the entry in Thycotic Secret Server.",
+            help_text="The secret-id used to select the entry in Delinea Secret Server.",
         )
         secret_selected_value = forms.ChoiceField(
             label="Return value",
             required=True,
-            choices=ThycoticSecretChoices,
+            choices=DelineaSecretChoices,
             help_text="Select which value to return.",
         )
 
 
-class ThycoticSecretServerSecretsProviderPath(ThycoticSecretServerSecretsProviderBase):
-    """A secrets provider for Thycotic Secret Server."""
+class DelineaSecretServerSecretsProviderPath(DelineaSecretServerSecretsProviderBase):
+    """A secrets provider for Delinea Secret Server."""
 
-    slug = "thycotic-tss-path"  # type: ignore
-    name = "Thycotic Secret Server by Path"  # type: ignore
+    slug = "delinea-tss-path"  # type: ignore
```

Comment:
This also is a breaking change and will invalidate any existing Secrets defined around the old slug. You *might* be able to add a data migration to the app that will auto-update existing Secrets.

Reply:
Added a migration to handle and update any pre-existing Thycotic secrets

**Pre-Upgrade:**
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/c9d3ab23-cba6-4bad-88c9-adabe047fb29)
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/834c2330-8b3b-4ef9-a4c4-36a4d4a8cf40)
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/a72bc050-f385-4f07-a12a-7b6d395ecc06)

**Post-Upgrade:**
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/d21c924f-0c30-4b63-87c1-bdd7bfac89fc)
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/f208c3b8-3c6f-4145-b92e-53962fbc6a26)
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/ebd72cf8-4d9c-4864-96b5-1e05f4bc747c)